### PR TITLE
Allow --unstable to be set with an environment variable

### PR DESCRIPTION
**Pull request by neunenak**
Allow the environment variable JUST_ALLOW_UNSTABLE=true to act as equivalent to the --unstable flag. Any other value of JUST_ALLOW_UNSTABLE is treated as false. 

This addresses https://github.com/casey/just/issues/1575

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -3,6 +3,8 @@ use {
   clap::{App, AppSettings, Arg, ArgGroup, ArgMatches, ArgSettings},
 };
 
+pub(crate) const UNSTABLE_KEY: &str = "JUST_ALLOW_UNSTABLE";
```
Comment (casey) [inline src/config.rs]: What about `JUST_UNSTABLE` instead? `JUST_UNSTABLE=true` makes sense, and is shorter.

File: src/config.rs [hunk 2]
```diff
@@ -569,6 +571,9 @@ impl Config {
       None
     };
 
+    let unstable = matches.is_present(arg::UNSTABLE)
+      || std::env::var_os(UNSTABLE_KEY).map_or(false, |val| val.to_string_lossy() == "true");
```
Comment (casey) [inline src/config.rs]: I'd be inclined to inline this:
```suggestion
      || std::env::var_os("JUST_UNSTABLE").map_or(false, |val| val.to_string_lossy() == "true");
```

**Review by casey — CHANGES_REQUESTED** [review]
Just a few random nitpicks, see comments. Also: Is there any sort of convention regarding how other tools convert environment variable values are converted to bools? For example, Rust has the `RUST_BACKTRACE` variable, which if unset or set to `0` will not capture a backtrace, but if set to any other value will. I wonder if there's any particular reason to go with one standard vs another.

**Comment by neunenak** [comment]
> Just a few random nitpicks, see comments. Also: Is there any sort of convention regarding how other tools convert environment variable values are converted to bools? For example, Rust has the `RUST_BACKTRACE` variable, which if unset or set to `0` will not capture a backtrace, but if set to any other value will. I wonder if there's any particular reason to go with one standard vs another.

I'm not aware of any convention other than what seems reasonable from the perspective of a developer working with standard programming languages. I'd probably allow both "0" and "false" to be treated as a false boolean, the convention that setting the env var to anything else makes it true, seems reasonable to me.

**Comment by casey** [comment]
Tweaked a little bit and merged. Sorry for letting this sit for so long. In the mean time, `--dump-format json` was stabilized, so I changed the test to use `--fmt` instead.

**Pull request merged**